java/src/main/java/org/whispersystems/signalservice/loki/api/LokiPublicChatAPI.kt

```diff
@@ -210,6 +210,19 @@ class LokiPublicChatAPI(private val userHexEncodedPublicKey: String, private val
         }
     }
 
+    public fun deleteMessages(messageServerIDs: LongArray, channel: Long, server: String, isSentByUser: Boolean): Promise<LongArray, Exception> {
+        return retryIfNeeded(maxRetryCount) {
+            val isModerationRequest = !isSentByUser
+            val parameters = mapOf( "ids" to messageServerIDs.joinToString() )
+            Log.d("Loki", "Deleting message with IDs: ${messageServerIDs.joinToString()} from public chat channel with ID: $channel on server: $server (isModerationRequest = $isModerationRequest).")
+            val endpoint = if (isSentByUser) "loki/v1/messages" else "loki/v1/moderation/messages"
+            execute(HTTPVerb.DELETE, server, endpoint, parameters = parameters).then {
+                Log.d("Loki", "Deleted message with ID: $messageServerIDs from public chat channel with ID: $channel on server: $server.")
+                messageServerIDs
+            }.get()
```

Comment:
**FUTURE NOTE:**
This hasn't been merged in to dev yet but we're slowly removing `.get()` calls inside `retryIfNeeded`. You won't need to update this PR with it as i'll go ahead and fix it up once multi-device gets merged. 

Essentially calling `.get` will block the thread and we should avoid that at all costs.